util/circular-chk.py

```diff
@@ -0,0 +1,51 @@
+import glob
+import re
+from sys import exit, argv
+from pathlib import Path
+
+strict = False
+if len(argv) > 1 and argv[1] == '--strict' :
+    strict = True
+
+regexExclude = r"keymaps\/(.*)via\/"
+reg = re.compile(regexExclude)
+filesToscan = Path('keyboards').glob('**/rules.mk')
+illegalSRC = list()
+
+for f in filesToscan:
+    p = str(f).replace('\\','/')
+    basepath = p.replace('keyboards/','')
+    basepath_arr = basepath.split('/')
+
+    # Skip 'via' file
+    if reg.search(p):
+        continue
+
+    # Read file as lines
+    lineList = list()
+    with f.open() as file:
+        try:
+            lineList = file.readlines()
+        except:
+            # sometimes encoding issues with japanese characters for example
+            pass
+
+    # Concat multilines
+    lineList = '\n'.join(lineList).replace('\\\n','').split('\n')
```

Comment:
You can replace all of this with `file_lines = file.read_text().replace('\\\n', '').split('\n')`. Also, if there is still a need for a try/except block do not silently eat all exceptions. Catch specific exceptions and/or log the error and stack trace.

Reply:
It was due to comments in japanese characters so it was revelant to swallow them instead of polluting the stdout. I can prompt it but it's a non blocking step.

Reply:
If there's a legit need to catch the exception that's fine. The problem is silently discarding all exceptions.